source/Octopus.Tentacle.Client/Scripts/HalibutClientExceptionExtensionMethods.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using Halibut;
+using Halibut.Exceptions;
+
+namespace Octopus.Tentacle.Client.Scripts
+{
+    public static class HalibutClientExceptionExtensionMethods
+    {
+        public static bool IsConnectingOrTransferringRequestCancelledException(this Exception ex)
+        {
+            return ex is ConnectingRequestCancelledException or TransferringRequestCancelledException ||
+                ex.IsHalibutWrappedConnectingRequestCancelledException() ||
+                ex.IsHalibutWrappedTransferringRequestCancelledException();
+        }
+
+        public static bool IsHalibutWrappedConnectingRequestCancelledException(this HalibutClientException? ex)
+        {
+            if (ex is null)
+            {
+                return false;
+            }
+
+            // TODO: Make this type safe
+            if (ex.Message.Contains("The Request was cancelled while Connecting"))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public static bool IsHalibutWrappedConnectingRequestCancelledException(this Exception ex)
+        {
+            return (ex as HalibutClientException).IsHalibutWrappedConnectingRequestCancelledException();
+        }
+
+        public static bool IsHalibutWrappedTransferringRequestCancelledException(this HalibutClientException? ex)
+        {
+            if (ex is null)
+            {
+                return false;
+            }
+
+            // TODO: Make this type safe
```

Comment:
We have tesats that will catch if this stops working but it needs to be improved. This will be done in a future PR